src/Microsoft.AspNetCore.Antiforgery/DefaultAntiforgeryTokenGenerator.cs

```diff
@@ -40,7 +41,17 @@ public AntiforgeryToken GenerateCookieToken()
                 throw new ArgumentNullException(nameof(httpContext));
             }
 
-            Debug.Assert(IsCookieTokenValid(cookieToken));
+            if (cookieToken == null)
+            {
+                throw new ArgumentNullException(nameof(cookieToken));
+            }
+
+            if (!IsCookieTokenValid(httpContext, cookieToken))
+            {
+                throw new ArgumentException(
+                    Resources.Antiforgery_CookieToken_MustBeProvided_Generic,
```

Comment:
The message `The required antiforgery cookie token must be provided.` is misleading here. Since we know the caller passed a non-`null` `cookieToken`, the only possible problem is that the caller passed a non-cookie `AntiForgeryToken`. Seems more like `AntiforgeryToken_SecurityTokenMismatch` or `AntiforgeryToken_TokensSwapped`.